cybersyn/scripts/layout.lua

```diff
@@ -82,6 +82,8 @@ function remove_train(map_data, train_id, train)
 	interface_raise_train_removed(train_id, train)
 end
 
+
+local size_fluidwagon_cache = {}
```

Comment:
I believe this will cause desyncs because the content is obtained by modifying game-state. Scenario: 
1. The session of player-1 already has cached info
1. Player-2 joins but still needs to cache the info
1. Next time the info is needed the games of player-1  and 2 will do different things.

I think the best place to store this would be `storage.perf_cache.size_fluidwagon`.
That is replicated game-state and avoids the problem.
perf_cache is already set up and wiped in `on_config_change`.